protocol/src/main/java/org/fido/iot/protocol/To0ServerService.java

```diff
@@ -58,6 +66,36 @@ protected void doOwnerSign(Composite request, Composite reply) {
       throw new InvalidOwnerSignBodyException();
     }
 
+    List<String> allowlistkeys = getStorage().getOvKeysAllowlist();
+    List<String> denylistkeys = getStorage().getOvKeysDenylist();
+
+    //Verifying Manufacturer Public key against allowlist and denylist
+    Composite publicKey = voucher.getAsComposite(Const.OV_HEADER).getAsComposite(Const.OVH_PUB_KEY);
+    PublicKey pub = cryptoService.decode(publicKey);
+    MessageDigest md = MessageDigest.getInstance(Const.SHA_256_ALG_NAME);
+    byte[] keyHash = md.digest(pub.getEncoded());
+    String hexHash = DatatypeConverter.printHexBinary(keyHash);
+
+    if (!cryptoService.verifyOvPublicKeyHash(hexHash, allowlistkeys, denylistkeys)) {
+      throw new InvalidOwnerSignBodyException();
+    }
+
+    //Verifying Owner public key against allowlist and denylist
```

Comment:
I thought we were going to just check the mfg key against the lists since checking the owner key as well would not be scalable?

Reply:
I thought it was still under discussion and the other alternative was either manufacturer or of the owner keys should be in the allow list.

Reply:
Implementation changed as per the agreed policy: No key in denylist and atleast one in allow list.